src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/LocalDBTest/LocalDBTest.cs

```diff
@@ -134,16 +165,25 @@ private static void OpenConnection(string connString)
 
         private static string GetLocalDbNamedPipe()
         {
-            string state = ExecuteLocalDBCommandProcess(s_commandPrompt, s_sqlLocalDbInfo, "state");
-            while (state.Equals("stopped", StringComparison.InvariantCultureIgnoreCase))
+            RestartLocalDB();
+            string instanceName = ExecuteLocalDBCommandProcess(s_commandPrompt, s_sqlLocalDbInfo, InfoType.pipeName);
+            Assert.NotNull(instanceName);
+            Assert.NotEmpty(instanceName);
+            return instanceName;
+        }
+
+        private static void RestartLocalDB()
+        {
+            string state = ExecuteLocalDBCommandProcess(s_commandPrompt, s_sqlLocalDbInfo, InfoType.state);
+            int count = 5;
+            while (state.Equals("stopped", StringComparison.InvariantCultureIgnoreCase) && count>0)
             {
-                state = ExecuteLocalDBCommandProcess(s_commandPrompt, s_startLocalDbCommand, "state");
+                count--;
+                state = ExecuteLocalDBCommandProcess(s_commandPrompt, s_startLocalDbCommand, InfoType.state);
                 Thread.Sleep(2000);
             }
```

Comment:
Regarding this [comment](https://github.com/dotnet/SqlClient/pull/1499#discussion_r798990435), you missed throwing the exception.